### PR TITLE
[MRG] CI consolidation + minor bug fixes

### PR DESCRIPTION
**Pull request by tgsmith61591**
# Description

* Circle CI changes:
  - Refactor Circle CI to build wheels during each unit test stage. This is in direct response to several issues we've had during the last several deployments, which were not discovered until the release was cut. Each unit test stage builds a wheel, runs the tests, and then persists the wheel to the workspace for the deployment stage. After merging, we will need to cut a new test release to ensure we are properly restoring the workspace.
  - Consolidate smaller, miscellaneous tests into a single stage to reduce the concurrency contention imposed by Circle CI's rate limits
* Address issue #440
* Speed up some tests & examples
* Add to documentation

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation change

# How Has This Been Tested?

Everything passes (still)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pmdarima/utils/visualization.py [hunk 1]
```diff
@@ -371,6 +372,15 @@ def tsdisplay(y, lag_max=50, figsize=(8, 6), title=None, bins=25,
     ax1 = fig.add_subplot(gs[2:, 0])
     ax2 = fig.add_subplot(gs[2:, 1])
 
+    # make sure y is a np array
+    y = check_endog(y, copy=False)
+
+    if lag_max >= y.shape[0]:
+        raise ValueError(
+            f"lag_max ({lag_max}) must be < than length of the "
```
Comment (aaronreidsmith) [inline pmdarima/utils/visualization.py:380]: Nit: `<` means "less than," so this reads as "lag_max (x) must be less than *than* length...". I think we should remove one or the other. I would do "less than" (with no symbol), but up to you

**Review by aaronreidsmith — APPROVED** [review]
LGTM. Not changing the title since you still have it marked as WIP, but looks good (with the exception of one nit) and seems to run faster!

File: pmdarima/utils/visualization.py (hunk 1 above)
Comment (tgsmith61591) [reply pmdarima/utils/visualization.py:380]: If only there were an eyeroll emoji. 😉

Comment (aaronreidsmith) [reply pmdarima/utils/visualization.py:380]: Like this? 🙄

Comment (tgsmith61591) [reply pmdarima/utils/visualization.py:380]: https://media.tenor.com/images/f8a02c67648240f3eba5b3fb871e7c37/tenor.gif

Comment (aaronreidsmith) [reply pmdarima/utils/visualization.py:380]: pepe reeeee

:trollface:

**Pull request merged**